skglm/utils/validation.py

```diff
@@ -23,3 +28,68 @@ def check_group_compatible(obj):
                 f"'{obj_name}' is not block-separable. "
                 f"Missing '{attr}' attribute."
             )
+
+
+def check_obj_solver_attr(obj, solver, required_attr, support_sparse=False):
+    """Check whether datafit or penalty is compatible with solver.
+
+    Parameters
+    ----------
+    obj : Instance of Datafit or Penalty
+        The instance Datafit (or Penalty) to check.
+
+    solver : Instance of Solver
+        The instance of Solver to check.
+
+    required_attr : List or tuple of strings
+        The attributes that ``obj`` must have.
+
+    support_sparse : bool, default False
+        If ``True`` adds a ``SPARSE_SUFFIX`` to check compatibility with sparse data.
+
+    Raises
+    ------
+        AttributeError
+            if any of the attribute in ``required_attr`` is missing
+            from ``obj`` attributes.
+    """
+    missing_attrs = []
+    suffix = SPARSE_SUFFIX if support_sparse else ""
+
+    # if `attr` is a list check that at least one of them
```

Comment:
why "at least one of them" ?

Reply:
ok got it thanks to your expalnations @Badr-MOUFAD , maybe explain with a bit more words here too